src/main/java/org/symphonyoss/symphony/messageml/elements/Form.java

```diff
@@ -14,6 +16,8 @@
 public class Form extends Element {
   public static final String MESSAGEML_TAG = "form";
   private static final String ID_ATTR = "id";
+
+  private static List<String> builtFormIds = new ArrayList<>();
```

Comment:
The static modifier is a problem. It means that, from the moment you start the Agent, each form in each message will have to have a different ID. However, if you restart the Agent, or if you start another one, this ID registry will reset and you will be able to send forms with same IDs as previously sent ones. This is a weirdly inconsistent and totally opaque behaviour where form IDs are neither unique per message nor globally unique in the whole system.

Let's illustrate this with code:

```
  @Test
  public void testMultipleIds() throws Exception {
    IDataProvider dataProvider = new TestDataProvider();
    MessageMLContext context = new MessageMLContext(dataProvider);

    String input = "<messageML><form id=\"form1\"></form><form id=\"form2\"></form></messageML>";

    context.parseMessageML(input, null, MessageML.MESSAGEML_VERSION);

    try {
      // A new message comes in...
      context.parseMessageML(input, null, MessageML.MESSAGEML_VERSION);
    } catch (InvalidInputException e) {
      System.out.println("Ooops!");
    }

    try {
      // Let's reinitialize the MML context...
      context = new MessageMLContext(dataProvider);
      context.parseMessageML(input, null, MessageML.MESSAGEML_VERSION);
    } catch (InvalidInputException e) {
      System.out.println("Ooops!");
    }
  }
```

Reply:
We do need the ids, and the document actually specifies it's required, you were probably seeing in the wrong place.

And this is standard HTML, on both the formId being able to be sent (and identify that form) and on the id being unique.